modules/renter/files.go

```diff
@@ -257,21 +258,30 @@ func (r *Renter) DeleteFile(nickname string) error {
 
 // FileList returns all of the files that the renter has.
 func (r *Renter) FileList() []modules.FileInfo {
+	// Get all the files and their contracts
 	var files []*file
+	contractIDs := make(map[types.FileContractID]struct{})
 	lockID := r.mu.RLock()
 	for _, f := range r.files {
 		files = append(files, f)
+		for cid := range f.contracts {
+			contractIDs[cid] = struct{}{}
+		}
 	}
 	r.mu.RUnlock(lockID)
```

Comment:
This is a lock that's potentially going to be held for a long time, if there are a lot of files to iterate through.

Reply:
I guess there isn't any super easy way to deal with it.